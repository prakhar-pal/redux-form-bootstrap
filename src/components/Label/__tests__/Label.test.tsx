import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Label } from '../index';
describe('Label',()=>{
    it('renders string',()=>{
        const app = render(<Label value="this is a label"/>);
        expect(app).toMatchSnapshot();
    })
    it('renders object',()=>{
        const app = render(<Label value={{htmlFor:'myLabel',value:'This is to be shown'}}/>);
        expect(app).toMatchSnapshot();
    })

    it('renders correct label and value when object is passed',()=>{
        const labelText = 'This is to be shown';
        const app = render(<Label value={{htmlFor:'myLabel',value: labelText}}/>);
        expect(app.getByText(labelText)).toBeInTheDocument();
        expect(document.querySelector('[for="myLabel"]')).toBeInTheDocument();
    });

    it('render text label', () => {
        const app = render(<Label value='myLabel'/>);
        expect(app.getByText('myLabel')).toBeInTheDocument();
        expect(document.querySelector('[for="myLabel"]')).toBeInTheDocument();
    })
});