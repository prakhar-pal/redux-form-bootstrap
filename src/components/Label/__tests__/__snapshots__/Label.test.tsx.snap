// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders object 1`] = `
<Label
  for="myLabel"
  onClick={[Function]}
  role="label"
  tag="label"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  This is to be shown
</Label>
`;

exports[`Label renders string 1`] = `
<Label
  for="this is a label"
  onClick={[Function]}
  role="label"
  tag="label"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  this is a label
</Label>
`;
